### PR TITLE
[Feature][Connector][Elasticsearch] Support https protocol

### PR DESCRIPTION
**Pull request by hailin0**
## Purpose of this pull request

* support https protocol
* compatible with opensearch
* 
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [x] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by hailin0** [comment]
PTAL

File: seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/util/DerUtils.java [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * ASN.1 DER encoder methods necessary to process PEM files and to write a certificate signing request.
+ */
+@SuppressWarnings("MagicNumber")
+public final class DerUtils {
+    public static final int SEQUENCE_TAG = 0x30;
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/util/DerUtils.java]: This class looks to be from the `airlift` project, please confirm.

Comment (hailin0) [reply seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/util/DerUtils.java]: fixed.   please review

File: seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/client/EsRestClient.java [hunk 2]
```diff
@@ -62,52 +69,115 @@ public class EsRestClient {
...
+                throw new RuntimeException(e);
+            }
+            return httpClientBuilder;
+        });
+        return restClientBuilder;
     }
 
     public BulkResponse bulk(String requestBody) {
-        Request request = new Request("POST", "_bulk");
+        Request request = new Request("POST", "/_bulk");
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/client/EsRestClient.java:180]: fix https://bugs.chromium.org/p/gerrit/issues/detail?id=9761

Comment (hailin0) [reply seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/client/EsRestClient.java:180]: compatible with aws opensearch

**Review by EricJoy2048 — COMMENTED** [review]
Please add release note to `/release-note.md`

**Comment by hailin0** [comment]
> Please add release note to `/release-note.md`

fixed @EricJoy2048

**Review by CalvinKirs — APPROVED** [review]

**Review by TyrantLucifer — APPROVED** [review]

**Comment by kpretty** [comment]
@hailin0 `airlift` depends on `guava` which seems to be incompatible with version 19.0 introduced by the parent project，exist `NoSuchMethodError`

**Pull request merged**